qiskit_nature/algorithms/initial_points/initial_point.py

```diff
@@ -0,0 +1,44 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The initial point interface."""
+
+from abc import ABC, abstractmethod
+
+import numpy as np
+
+from qiskit_nature.properties.second_quantization import GroupedSecondQuantizedProperty
+from qiskit_nature.circuit.library import UCC
+
+
+class InitialPoint(ABC):
```

Comment:
So I am wondering if/how this might extend to UVCC+VQE. This seems to be generic in its name yet I guess it ends up more specific to UCC? I am thinking of #571 and how that relates to UVCC+VQE factory if/when EvolveOpAnsatz drops the preferred point logic. As in does this InitialPoint class concept extend to there too; should any naming etc here perhaps accommodate for that. Just putting this out there for feedback from others too.

Reply:
@woodsp-ibm, should I open a separate issue to review this, or document it in #571 or #632 to get more eyes on it? I suppose, in principle, any MinimumEigensolverFactory using any evolved ansatz circuit could be provided with some InitialPoint that provides a suitable ndarray on request, but, I have currently just set it up for UCC. I suppose the options, as you've already laid out, are:

1. Open up InitialPoint to be used with any MinimumEigensolverFactory instance, and EvolvedOperator ansatz  (I don't know how useful this will be in practice.
2. Create a parent class for both UCC+VQE and UVCC+VQE, such that they inherit this functionality, and refactor InitialPoint to expect either a UCC or UVCC ansatz (or a new interface).
3. Rename InitialPoint to UCCInitialPoint or similar.

Reply:
@declanmillar Sorry, I meant to already reply here. I think it makes sense to keep the UCC and UVCC stacks aligned. I already talked to @paulineollitrault and we discussed that the "HF-equivalent" in the vibrational stack should be called `VSCF` (like the in in the initial state library).

Reply:
In terms of support UVCC+VQE, I think at present it would be limited to having an equivalent to the HF initial point as Max describes above. Allowing other points to be developed, though we would not be including any (beyond what you are doing with UCC and MP2). For UVCC this would be another all zeros point. (This will then allow the removal of the preferred initial point all zeros from the EvolvedOpAnstaz base class, and VQE's use of that, relying instead on explicit initial point setting as per what you have done so far.) As to how to accommodate this type of initial point, maybe in the 632 PR as you suggest, which may end up renaming/refactoring some of these classes a little to broaden things out. The reason these factories exist is that the ansatz/initial state need configuring with information from the problem, so its not needed for any ansatz per-se. UVCC+VQE is already "similar" but I am not sure that whether a common parent, or perhaps mixin, is needed. Maybe things will become more evident, either way, when its worked out further. I would be wary though of trying to generalize things too much at this point - I was really thinking more of consistency around UCC and UVCC factories in regards of the initial point handling at this stage. Things can always be refactored later in the future if more generality is needed if perhaps we then have more cases to consider etc.

Reply:
I've added a note to consider this as part of PR #632.